src/elements/element.line.js

```diff
@@ -52,57 +33,104 @@ function setStyle(ctx, vm) {
 	ctx.strokeStyle = vm.borderColor;
 }
 
-function normalPath(ctx, points, spanGaps, vm) {
-	const steppedLine = vm.steppedLine;
-	const lineMethod = steppedLine ? helpers.canvas._steppedLineTo : helpers.canvas._bezierCurveTo;
-	let move = true;
-	let index, currentVM, previousVM;
+function lineTo(ctx, previous, target) {
+	ctx.lineTo(target.x, target.y);
+}
+
+function getLineMethod(line) {
+	if (line.steppedLine) {
+		return _steppedLineTo;
+	}
+
+	if (line.tension) {
+		return _bezierCurveTo;
+	}
+
+	return lineTo;
+}
+
+function getInterpolationMethod(line) {
+	if (line.steppedLine) {
+		return _steppedInterpolation;
+	}
 
-	for (index = 0; index < points.length; ++index) {
-		currentVM = points[index]._view;
+	if (line.tension) {
+		return _bezierInterpolation;
+	}
 
-		if (currentVM.skip) {
-			move = move || !spanGaps;
+	return _pointInLine;
+}
+
+function pathSegment(ctx, line, segment, options) {
+	const {start, end, loop} = segment;
+	const {points, _view: vm} = line;
+	const lineMethod = getLineMethod(vm);
+	const count = points.length;
+	let {move = true, reverse} = options || {};
```

Comment:
`move` can't be specified in the options, can it? If that's the case, I think it'd make that clearer to declare it on its own line

Reply:
It can and is, from filler.

Reply:
Is it documented anywhere? I didn't see it on https://www.chartjs.org/docs/latest/charts/area.html. Could we add it?

Reply:
Oh, there is a misunderstanding here. Maybe I should rename `options` to `args` here, these are just options passed to the function, not user provided.

Reply:
Thanks for clarifying. Yeah, `args` or `params` or something else would be good